src/main/software/amazon/event/ruler/ByteMachine.java

```diff
@@ -810,9 +810,13 @@ NameState findPattern(final Patterns pattern) {
 
     private NameState findAnythingButPattern(AnythingBut pattern) {
 
-        Set<NameState> nextNameStates = pattern.getValues().stream().
-                map(value -> findMatchPattern(getParser().parse(pattern.type(), value), pattern)).
-                filter(Objects::nonNull).collect(Collectors.toSet());
+        Set<NameState> nextNameStates = new HashSet<>(pattern.getValues().size());
+        for (String value : pattern.getValues()) {
+            NameState matchPattern = findMatchPattern(getParser().parse(pattern.type(), value), pattern);
+            if (Objects.nonNull(matchPattern)) {
```

Comment:
Um, the javadocs say that `Objects.nonNull()` is designed to be used while filtering.  Obviously it does no harm, but when I saw this I had to go read a bunch of stuff to figure out whether this was better for some cosmic reason than just `matchPattern != null` and it's not.  So maybe let's not make other people get the same puzzled look on their face and and up at StackOverflow…

Reply:
jajaja, I got biased by the streams (very common in that context) , I did try to make it closer to the other side, but I think you are right,  != null is crystal clear